### PR TITLE
[TreeView] Create the package on the X repository

### PR DESCRIPTION
**Pull request by flaviendelangle**
## In this PR

- Migrate the codebase
- Set up the doc and prop-types generation
- Migrate to TS

## Follow up

- Migrate the doc
- Remove from the lab (with a warning message)
- Write codemod

### CODE REVIEW

File: packages/x-tree-view/README.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
+# @mui/x-charts
```
Comment (noraleonte) [inline packages/x-tree-view/README.md:1]: Is this right? 
```suggestion
# @mui/x-tree-view
```

File: packages/x-tree-view/package.json [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+  "name": "@mui/x-tree-view",
+  "version": "6.0.0-alpha.0",
+  "description": "The community edition of the tree view components (MUI X).",
+  "author": "MUI Team",
+  "main": "src/index.ts",
+  "license": "MIT",
+  "bugs": {
+    "url": "https://github.com/mui/mui-x/issues"
+  },
+  "homepage": "https://mui.com/x/react-date-tree-view/",
```
Comment (noraleonte) [inline packages/x-tree-view/package.json:11]: This looks off to me 🤔

**Review by noraleonte — COMMENTED** [review]
Looking great! 🎉 
I left two very small comments, I think they might be small oversights from the migration 🤔

File: packages/x-tree-view/README.md (hunk 1 above)
Comment (flaviendelangle) [reply packages/x-tree-view/README.md:1]: Nice catch !

File: packages/x-tree-view/package.json (hunk 2 above)
Comment (flaviendelangle) [reply packages/x-tree-view/package.json:11]: Yes, the doc is not live yet, we will have to publish it before actually releasing the package :+1:

Comment (noraleonte) [reply packages/x-tree-view/package.json:11]: But will it be 'react-date-tree-view'?

Comment (flaviendelangle) [reply packages/x-tree-view/package.json:11]: Oh you are right :laughing:

**Pull request merged**